companion/src/helpers.h

```diff
@@ -162,4 +162,31 @@ QSet<QString> getFilesSet(const QString &path, const QStringList &filter, int ma
 
 bool caseInsensitiveLessThan(const QString &s1, const QString &s2);
 
+
+class gpsGlitchFilter
+{
+public:
+  gpsGlitchFilter() : lastValid(false), glitchCount(0) {};
+  bool isGlitch(double latitude, double longitude);
+
+private:
+  bool lastValid;
+  int glitchCount;
+  double lastLat;
+  double lastLon;
+};
+
+class gpsLatLonFilter
+{
+public:
+  gpsLatLonFilter() {};
+  bool isValid(QString latitude, QString longitude);
```

Comment:
Those args should be const QString & (unless they are modified). It will avoid a couple of memcpy ...